### PR TITLE
Fix idp-htpasswd for mgmt CI

### PR DESCRIPTION
**Pull request by xingxingxia**
@heliubj18 @LiangquanLi930 , how about use this PR? Replacing https://github.com/openshift/release/pull/54651 . CC @fxierh

### CODE REVIEW

File: ci-operator/step-registry/idp/htpasswd/idp-htpasswd-commands.sh [hunk 1]
```diff
@@ -6,13 +6,14 @@ set -o pipefail
...
 function check_if_hypershift_env () {
-    if [ -f "${SHARED_DIR}/nested_kubeconfig" ]; then
-        IS_HYPERSHIFT_ENV="yes"
-    else
-        # We must set IS_HYPERSHIFT_ENV="no" otherwise OCP CI will fail because this script sets "set -u".
-        IS_HYPERSHIFT_ENV="no"
+    if [ ! -f "${SHARED_DIR}/cluster-type" ] || [ $(cat "${SHARED_DIR}/cluster-type") != "hypershift-guest" ] ; then
+        echo "This is not a HyperShift guest cluster CI job."
+        # We must set IS_HYPERSHIFT_GUEST_ENV="no" otherwise OCP CI will fail in "set_common_variables" later because this script sets "set -u".
+        IS_HYPERSHIFT_GUEST_ENV="no"
```
[the remark this answers is not in the document]
Comment (xingxingxia) [reply ci-operator/step-registry/idp/htpasswd/idp-htpasswd-commands.sh:12]: Checked the definition of "aws-ipi-ovn-hypershift-fips-mgmt-f14" jobs. In the past, they performed it on the management cluster. This PR does not change that fact.

**Comment by LiangquanLi930** [comment]
@liangxia for https://github.com/openshift/release/pull/54651/files, I can confirm that the modification is correct. For Hypershift, currently, all steps will have nested_kubeconfig. Therefore:
```
if [ -f "${SHARED_DIR}/nested_kubeconfig" ]; then
    IS_HYPERSHIFT_ENV="yes"
else
    # We must set IS_HYPERSHIFT_ENV="no" otherwise OCP CI will fail because this script sets "set -u".
    IS_HYPERSHIFT_ENV="no"
    return 0
fi
```
There is no issue with this approach. 

Hypershift CI is divided into guest and management types. Since guest CI requires switching kubeconfig, we generate mgmt_kubeconfig. For mgmt CI, there's no need to switch kubeconfig, so the current kubeconfig is already the mgmt kubeconfig.

So, the logic is:
```
elif [ -f "${SHARED_DIR}/mgmt_kubeconfig" ]; then
    export KUBECONFIG="${SHARED_DIR}/mgmt_kubeconfig"
else
    export KUBECONFIG="${SHARED_DIR}/kubeconfig"
fi
```
This is why I needed to make this modification.

**Comment by xingxingxia** [comment]
@LiangquanLi930 I knew that. But I commented and explained details that you need to set one variable to "no" after you do `export KUBECONFIG="${SHARED_DIR}/kubeconfig"`. Thus I directly submit all changes here.

**Comment by xingxingxia** [comment]
@heliubj18 @LiangquanLi930 @fxierh I sent you a meeting invitation to discuss it.

**Comment by xingxingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-ovn-hypershift-mgmt-critical-f28

**Comment by xingxingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-guest-critical-f28

**Comment by xingxingxia** [comment]
@heliubj18 tested mgmt CI aws-ipi-ovn-hypershift-mgmt-critical-f28, original failure is gone / fixed. Also ran regression test with other CI aws-ipi-ovn-hypershift-guest-critical-f28 , no issue. If looks good to you, please /approve.

**Comment by xingxingxia** [comment]
After sync, we concluded:
1 hypershift management CI forever will not run cucushift cases, therefore does not care about whether setting up idp or not. In terms of this, it is good to even get rid of the `idp-htpasswd` step in hypershift management CI.
2 Even if hypershift management CI will have to set up idp in future, hypershift qe team would like to switch to set up idp in the guest cluster. So they will use https://github.com/openshift/release/pull/54828 for it.
So closing this:
/close

**Pull request closed without merging**